model/project_parser_db.go

```diff
@@ -73,15 +76,21 @@ func (pp *ParserProject) UpsertWithConfigNumber(num int) error {
 	}
 	if found == nil {
 		pp.ConfigUpdateNumber = num
+		// this could error but it seems unlikely
 		return errors.Wrap(pp.Insert(), "error inserting parser project")
 	}
 
-	return errors.Wrapf(ParserProjectUpdateOne(
+	// NOTE: if $lt: num is true this will error which we don't want ($lt bc we may want to update other things still if equal)
+	// either we case here on ErrorNotFound and continue (it's fine if LoadProjectForVersion is using an older version bc that's what it did before so)
+	// or we check before updating if $lte but potentially racey
+	err = ParserProjectUpdateOne(
 		bson.M{
-			ParserProjectIdKey: pp.Id,
-			"$or": []bson.M{
-				bson.M{ParserProjectConfigNumberKey: bson.M{"$exists": false}},
-				bson.M{ParserProjectConfigNumberKey: pp.ConfigUpdateNumber},
+			"$and": []bson.M{
```

Comment:
Isn't the $and implicit?

Reply:
yes haha whoops, was messing around with queries and forgot to remove